### ISSUE TITLE
MarkdownEditor duplicates on repeater reorder

### ISSUE THREAD

**Issue by njxqlus** [laravel-filament/filament#7829]
### Package

filament/filament

### Package Version

v3.0.24

### Laravel Version

10.19.0

### Livewire Version

v3.0.0-beta.9

### PHP Version

8.1

### Problem description

MarkdownEditor content is duplicated when reordering parent Repeater both using buttons or drag&drop. 

<img width="496" alt="Screenshot 2023-08-17 at 11 38 01" src="https://github.com/filamentphp/filament/assets/15072672/b013bde5-d0db-41f4-a345-850bc01c64ca">

https://github.com/filamentphp/filament/assets/15072672/6c6fe5b8-f45f-4e51-9e7d-359345f896d5

---

Please use **markdown-editor-bug** branch in reproduction repository.

In continuation #7485 

### Expected behavior

Same as RichEditor - no content/editor duplication on repeater reorder

### Steps to reproduce

```php
return $form
    ->schema([
        Forms\Components\Repeater::make('foo')->schema([
            Forms\Components\MarkdownEditor::make('bar')
        ])->reorderableWithButtons()
    ]);
```

### Reproduction repository

https://github.com/njxqlus/filamentphp-bug-report/tree/markdown-editor-bug

**Comment by mikewink** [comment]
I have the same problem. It also happens when using the delete action or when using drag and drop. With each drag, you will end up with one more editor in the form field dragged and dragged over.

**Comment by njxqlus** [comment]
Same with the Builder

**Comment by zircuitz** [comment]
Is there a fix for this on the horizon? Not complaining, but it just gets a little frustrating to use.

**Comment by RobertsG** [comment]
Any updates on this?

**Comment by njxqlus** [comment]
> Any updates on this?

Fixed in [3.1](https://github.com/filamentphp/filament/releases/tag/v3.1.0)
